src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -4111,16 +4168,66 @@ def create_app_service_plan(cmd, resource_group_name, name, is_linux, hyper_v, p
+                                            vnet_sub_id=subnet_info["subnet_subscription_id"])
+        _vnet_delegation_check(cmd, subnet_subscription_id=subnet_info["subnet_subscription_id"],
+                               vnet_resource_group=subnet_info["resource_group_name"],
+                               vnet_name=subnet_info["vnet_name"],
+                               subnet_name=subnet_info["subnet_name"])
+        subnet_resource_id = subnet_info["subnet_resource_id"]
+    else:
+        subnet_resource_id = None
+
+    if assign_identities is not None:
+        _, _, user_assigned_identities, enable_system_assigned_identity = _build_identities_info(assign_identities)
+    else:
+        user_assigned_identities = None
+        enable_system_assigned_identity = None
+
+    # Transform default_identity parameter into the proper structure
+    plan_default_identity = _build_plan_default_identity(default_identity)
+
+    poller = AppServicePlanCreate(cli_ctx=cmd.cli_ctx)(command_args={
+        "name": name,
+        "resource_group": resource_group_name,
+        "location": location,
+        "tags": tags,
+        "sku": sku_def.__dict__,
+        "reserved": plan_def.reserved,
+        "hyper_v": plan_def.hyper_v,
+        "per_site_scaling": plan_def.per_site_scaling,
+        "hosting_environment_profile": plan_def.hosting_environment_profile,
+        "async_scaling_enabled": plan_def.async_scaling_enabled,
+        "zone_redundant": zone_redundant if zone_redundant else None,
+        "is_custom_mode": is_managed_instance,
+        "network": {
+            "virtual_network_subnet_id": subnet_resource_id,
+        } if subnet_resource_id else None,
+        "rdp_enabled": rdp_enabled,
+        "mi_system_assigned": str(enable_system_assigned_identity) if enable_system_assigned_identity else None,
+        "mi_user_assigned": user_assigned_identities,
+        "plan_default_identity": plan_default_identity,
+        "registry_adapters": registry_adapters,
+        "install_scripts": install_scripts,
+        "storage_mounts": storage_mounts,
+    })
+
     if no_wait:
-        return sdk_no_wait(no_wait, client.app_service_plans.begin_create_or_update, name=name,
-                           resource_group_name=resource_group_name, app_service_plan=plan_def)
+        return poller
 
-    poller = client.app_service_plans.begin_create_or_update(resource_group_name, name, plan_def)
+    # Check if this is an asynchronous operation
+    is_async = is_async_response(poller)
 
-    # Only use progress bar for actual long-running operations (async scaling)
-    # Check if the operation is actually async by looking at the poller status
-    if poller.done():
-        # This completed synchronously (200 response), no need for progress bar
+    if not is_async:
+        # for synchronous operations, or if we are unable to get the initial response, directly return poller result
```

Comment:
I’m not really sure why we need to distinguish whether the response is async or not. In fact, LongRunningOperation is used to wait for the operation to finish and retrieve the result.